samples/2048.cr

```diff
@@ -3,24 +3,26 @@
 require "colorize"
 
 module Screen
+  alias Color = Colorize::ColorANSI
+
   TILES = {
-        0 => {:white, nil},
-        2 => {:black, :white},
-        4 => {:blue, :white},
-        8 => {:black, :yellow},
-       16 => {:white, :red},
-       32 => {:black, :red},
-       64 => {:white, :magenta},
-      128 => {:red, :yellow},
-      256 => {:magenta, :yellow},
-      512 => {:white, :yellow},
-     1024 => {:white, :yellow},
-     2048 => {:white, :yellow},
-     4096 => {:white, :black},
-     8192 => {:white, :black},
-    16384 => {:white, :black},
-    32768 => {:white, :black},
-    65536 => {:white, :black},
+        0 => {Color::White, nil},
+        2 => {Color::Black, Color::White},
+        4 => {Color::Blue, Color::White},
+        8 => {Color::Black, Color::Yellow},
+       16 => {Color::White, Color::Red},
+       32 => {Color::Black, Color::Red},
+       64 => {Color::White, Color::Magenta},
+      128 => {Color::Red, Color::Yellow},
+      256 => {Color::Magenta, Color::Yellow},
+      512 => {Color::White, Color::Yellow},
+     1024 => {Color::White, Color::Yellow},
+     2048 => {Color::White, Color::Yellow},
+     4096 => {Color::White, Color::Black},
+     8192 => {Color::White, Color::Black},
+    16384 => {Color::White, Color::Black},
+    32768 => {Color::White, Color::Black},
+    65536 => {Color::White, Color::Black},
```

Comment:
You can leave the symbols by adding ` of Int32 => {Color, Color}` to this hash declaration.

Reply:
Oh, right!

Reply:
Hmm, no it doesn't really seem to work:
```cr
  TILES = {
        0 => {:white, nil},
        2 => {:black, :white},
        4 => {:blue, :white},
        8 => {:black, :yellow},
       16 => {:white, :red},
       32 => {:black, :red},
       64 => {:white, :magenta},
      128 => {:red, :yellow},
      256 => {:magenta, :yellow},
      512 => {:white, :yellow},
     1024 => {:white, :yellow},
     2048 => {:white, :yellow},
     4096 => {:white, :black},
     8192 => {:white, :black},
    16384 => {:white, :black},
    32768 => {:white, :black},
    65536 => {:white, :black},
  } of Int32 => {Colorize::ColorANSI, Colorize::ColorANSI?}
```
```
Showing last frame. Use --error-trace for full trace.

In samples/2048.cr:6:11

 6 | TILES = {
             ^-
Error: no overload matches 'Hash(Int32, Tuple(Colorize::ColorANSI, Colorize::ColorANSI | Nil))#[]=' with types Int32, Tuple(Symbol, Nil)

Overloads are:
 - Hash(K, V)#[]=(key : K, value : V)
```

Reply:
That's weird... it works fine without tuples, see https://carc.in/#/r/7nts

Looks like it might be a bug /cc @asterite

Reply:
I created #8333.